src/main/java/com/example/ajaxonboardingproject/service/impl/MovieSessionServiceImpl.kt

```diff
@@ -13,25 +16,23 @@ class MovieSessionServiceImpl(
     override fun findAvailableSessions(
         movieId: String,
         date: LocalDateTime
-    ): List<MovieSession> {
+    ): Flux<MovieSession> {
         return movieSessionRepository.findByMovieIdAndShowTimeAfter(movieId, date)
     }
 
-    override fun add(session: MovieSession): MovieSession {
+    override fun add(session: MovieSession): Mono<MovieSession> {
         return movieSessionRepository.save(session)
     }
 
-    override fun get(id: String): MovieSession {
+    override fun get(id: String): Mono<MovieSession> {
         return movieSessionRepository.findById(id)
```

Comment:
bring back NoSuchElementException just make it reactive

Reply:
Done!